File: java/maximumAverageSubarray.java
class Solution {
    public double findMaxAverage(int[] nums, int k) {
        //get average of first k elements
        double currentAverage = 0;
        for (int i = 0; i < k; i++) {
            currentAverage += nums[i];
        }
        currentAverage /= k;
        //keep track of maximum average
        double maxAverage = currentAverage;
        for (int i = k; i < nums.length; i++) {
            //obtain the next current average by adding next element and removing first element
            currentAverage -= ((double) nums[i - k]) / k;
            currentAverage += ((double) nums[i]) / k;
            maxAverage = Math.max(maxAverage, currentAverage);
        }
        return maxAverage;
    }
}

File: python/maximum_average_subarray.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        #get the first average
        first_k_sum = 0
        for i in range(k):
            first_k_sum += nums[i]
        cur_average = first_k_sum / k
        max_average = cur_average
        for i in range(k, len(nums)):
            #get the new average by removing the first element and adding newest
            cur_average = cur_average - nums[i - k] / k + nums[i] / k
            max_average = max(cur_average, max_average)
        return max_average